mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/tasks/SharedThreadPool.kt

```diff
@@ -0,0 +1,20 @@
+package org.modelix.mps.sync.tasks
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.Service
+import org.modelix.kotlin.utils.UnstableModelixFeature
+import java.util.concurrent.Executors
+
+@UnstableModelixFeature(
+    reason = "The new modelix MPS plugin is under construction",
+    intendedFinalization = "This feature is finalized when the new sync plugin is ready for release.",
+)
+@Service(Service.Level.APP)
+class SharedThreadPool : Disposable {
+
+    val threadPool = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors())
```

Comment:
thought: I am not sure, if:

* we need to create our own thread pool.
* the thread pool should have the size `.availableProcessors()`
* we should share the thread pool between projects.
   A bug caused by one project could disrupt the synchronization of other plugins.

---

However, this is just a refactoring and is implemented like it worked before.